R/config.R

```diff
@@ -0,0 +1,250 @@
+character_or_null <- S7::new_union(S7::class_character, NULL)
+
+#' Exclusions Class
+#'
+#' Represents exclusion criteria for the pipeline.
+#'
+#' @param path A string specifying the path to a CSV file containing exclusion
+#' data. It should include at least the columns: `reference_date`,
+#' `report_date`, ' `state_abb`, `disease`.
+#' @param blob_storage_container Optional. The name of the blob storage
+#' container to get it from. If NULL, will look locally.
+#' @export
+Exclusions <- S7::new_class(
+  "Exclusions",
+  properties = list(
+    path = character_or_null,
+    blob_storage_container = character_or_null
+  )
+)
+
+#' GenerationInterval Class
+#'
+#' Represents the generation interval parameters.
+#'
+#' @param path A string specifying the path to the generation interval CSV file.
+#' @param blob_storage_container Optional. The name of the blob storage
+#' container to get it from. If NULL, will look locally.
+#' @export
+GenerationInterval <- S7::new_class(
+  "GenerationInterval",
+  properties = list(
+    path = character_or_null,
+    blob_storage_container = character_or_null
+  )
+)
```

Comment:
Can these inherit from some kind of superclass? Does S7 do that?

Reply:
I think it would be good to DRY up. The general pattern I'm trying to do here is:

config -> load azure creds -> look for ABS container -> if yes, download to `path` -> read from `path`

So every non-config file (data, params, exclusions) should follow this same pattern.

Reply:
Added